### PR TITLE
Update proxy

### PR DESCRIPTION
**Pull request by miiu96**
- Updated proxy dependencies
- Small refactoring on the faucet part
- Updated go version

### CODE REVIEW

File: api/groups/baseValidatorGroup.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ type validatorGroup struct {
 	*baseGroup
 }
 
-// NewNodeGroup returns a new instance of nodeGroup
+// NewValidatorGroup returns a new instance of nodeGroup
```
Comment (bogdan-rosianu) [inline api/groups/baseValidatorGroup.go:16]: validatorGroup*

File: faucet/privateKeysLoader.go [hunk 2]
```diff
@@ -58,7 +58,7 @@ func (pkl *PrivateKeysLoader) PrivateKeysByShard() (map[uint32][]crypto.PrivateK
 	}
 
 	for _, privKeyBytes := range privKeysBytes {
-		privKeyBytes, err := hex.DecodeString(string(privKeyBytes))
+		privKeyBytes, err = hex.DecodeString(string(privKeyBytes))
```
Comment (bogdan-rosianu) [inline faucet/privateKeysLoader.go:61]: please rename `privKeyBytes` as it is the same with the var name in the for loop

File: go.mod [hunk 3]
```diff
@@ -1,21 +1,62 @@
...
+	github.com/gin-gonic/gin v1.7.6
 	github.com/pkg/errors v0.9.1
 	github.com/stretchr/testify v1.7.0
 	github.com/urfave/cli v1.22.5
 	gopkg.in/go-playground/validator.v8 v8.18.2
 )
 
-replace github.com/ElrondNetwork/arwen-wasm-vm/v1_3 v1.3.19 => github.com/ElrondNetwork/arwen-wasm-vm v1.3.20-0.20210702122719-c891907234fa
+require (
+	github.com/ElrondNetwork/concurrent-map v0.1.3 // indirect
```
Comment (bogdan-rosianu) [inline go.mod:23]: are these needed? if you remove them, run go mod tidy again, will they disappear?

File: facade/baseFacade.go [hunk 4]
```diff
@@ -263,31 +257,21 @@ func (epf *ElrondProxyFacade) SendUserFunds(receiver string, value *big.Int) err
...
+	if err != nil {
+		return nil, err
 	}
 
-	version, ok := netConf[core.MetricMinTransactionVersion].(float64)
-	if !ok {
-		return nil, errors.New("cannot get version. something went wrong")
-	}
+	networkCfg := &data.NetworkConfig{}
+	err = json.Unmarshal(networkConfigBytes, networkCfg)
```
Comment (andreibancioiu) [inline facade/baseFacade.go:272]: Optional: see comment above. Returning a struct in `GetNetworkConfigMetrics` would be simpler.

File: cmd/proxy/main.go [hunk 5]
```diff
@@ -511,8 +515,8 @@ func createVersionsRegistry(
 		pubKeyConverter,
 		hasher,
 		marshalizer,
-		ecConf.FeeSettings.MaxGasLimitPerBlock,
-		ecConf.FeeSettings.MaxGasLimitPerMetaBlock,
+		ecConf.FeeSettings.GasLimitSettings[0].MaxGasLimitPerBlock,
+		ecConf.FeeSettings.GasLimitSettings[0].MaxGasLimitPerMetaBlock,
```
[the remark this answers is not in the document]
Comment (miiu96) [reply cmd/proxy/main.go:519]: removed because is no longer used.

File: faucet/privateKeysLoader.go (hunk 2 above)
Comment (miiu96) [reply faucet/privateKeysLoader.go:61]: done.

File: go.mod (hunk 3 above)
Comment (miiu96) [reply go.mod:23]: these were generated when I updated the go version

File: data/api.go [hunk 6]
```diff
@@ -13,6 +13,14 @@ type GenericAPIResponse struct {
 	Code  ReturnCode  `json:"code"`
 }
 
+// NetworkConfig in a dto that will keep information about the network config
```
[the remark this answers is not in the document]
Comment (miiu96) [reply data/api.go:16]: done.

**Review by andreibancioiu — APPROVED** [review]

File: cmd/proxy/main.go [hunk 7]
```diff
@@ -229,14 +229,18 @@ func initializeLogger(ctx *cli.Context) (nodeFactory.FileLoggingHandler, error)
...
+			LogFilePrefix:   logFilePrefix,
+		})
 		if err != nil {
 			return nil, fmt.Errorf("%w creating a log file", err)
 		}
 	}
 
 	if !check.IfNil(fileLogging) {
-		err = fileLogging.ChangeFileLifeSpan(time.Second * time.Duration(logFileLifeSpanInSec))
+		err = fileLogging.ChangeFileLifeSpan(time.Second*time.Duration(logFileLifeSpanInSec), 1024)
```
[the remark this answers is not in the document]
Comment (miiu96) [reply cmd/proxy/main.go]: added

**Review by bogdan-rosianu — DISMISSED** [review]

**Review by schimih — DISMISSED** [review]

**Review by schimih — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**